tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -1920,13 +1928,60 @@ static llvm::Value* GetStartOffsetX(const KernelMappingScheme& mapping_scheme,
                                     llvm::Value* thread_id_x,
                                     llvm::Type* index_ty,
                                     llvm::IRBuilder<>* b) {
-  if (mapping_scheme.DilatedX()) {
+  auto constant = [&](int64 val) {
+    return llvm::ConstantInt::get(index_ty, val);
+  };
+  if (mapping_scheme.GetIndexingOrder() == kStridedIndexingX) {
     return thread_id_x;
+  } else if (mapping_scheme.GetIndexingOrder() == kLinearStridedIndexingX) {
+    return b->CreateMul(thread_id_x, constant(mapping_scheme.GetVectorSize()));
   }
+  CHECK_EQ(mapping_scheme.GetIndexingOrder(), kLinearIndexingX);
   int64 x_num_steps =
       mapping_scheme.GetTileSizeX() / mapping_scheme.GetNumThreadsX();
-  return b->CreateMul(thread_id_x,
-                      llvm::ConstantInt::get(index_ty, x_num_steps));
+  return b->CreateMul(thread_id_x, constant(x_num_steps));
+}
+
+// This function calls emit_elem_function() x_num_steps times.  If
+// vector_size==1, then each element index passed to
+// emit_elem_function() will be separated by step_x. If vector_size>1,
+// then it must be a multiple of x_num_steps.  In that case, it
+// triggers a different indexing order that is vectorizable by
+// LLVM. It generates many groups of calls to emit_elem_function. Each
+// group is separated by step_x elements.  Inside a group, elements
+// are consecutive. If check_x_tile_bounds is true, then it will check
+// if the element index is in bound compared to tile_width before
+// calling emit_elem_function.
+static void Unroll(
```

Comment:
We need a more self-descriptive name than `Unroll`, perhaps `UnrollInnerTileLoopForVectorization`?

Reply:
I renamed it to UnrollInnerTileLoop, it doesn't always vectorize, it is an option. I find strange to specify an option in the name.